ha/util/consul_kv_store.py

```diff
@@ -122,7 +145,16 @@ def set(self, key: str, val: str=None):
         self._verify_data(key)
         if self.key_exists(key):
```

Comment:
How is this `key_exists` exception is expected to be handled by the user? Consul wise, if the kv already exists then it will be simply overwritten by new data. If you don't want to overwrite then you can instead do a CAS transaction with `cas=0`, with this, if key already exists, kv put operation will automatically fail and it will save a get request effort.

Reply:
Hey, I checked all the calls caller calling key_exists() or get to check if it already exists.
So looks like we can remove this call and exception both.

Reply:
Okay, do you expect lot of work for this to be done in the ongoing PR?

Reply:
will fix in: https://jts.seagate.com/browse/CORTX-31319